ksonnet-gen/ksonnet/emit.go

```diff
@@ -423,7 +424,7 @@ func (ao *apiObject) emit(m *indentWriter) {
 	for _, pm := range ao.properties.sortAndFilterBlacklisted() {
 		// Skip special properties and fields that `$ref` another API
 		// object type, since those will go in the `mixin` namespace.
-		if isSpecialProperty(pm.name) || pm.ref != nil {
+		if isSpecialProperty(pm.name) || isMixinRef(pm.ref) {
```

Comment:
A mixin is a programming languages term of art that refers to an inheritance-like object model, except it's strictly more powerful, and based in composition rather than declared hierarchies of inheritance. It gets its name because the members of one object are "mixed" into another without needing to say "x extends y" as you would in Java.